### ISSUE TITLE
Special characters in product numbers are not encoded in the URL

### ISSUE THREAD

**Issue by jwe-cdn** [SAP/cloud-sdk-js#1899]
**Describe the bug**
On using SAP Cloud SDK for calling for example the endpoint GET /sap/opu/odata/sap/API_PRODUCT_SRV/A_Product(Product='...') you can not use product numbers with special characters like '?'.
You get always an "Invalid URI segment" error.
As far as I know, this worked in older versions.
To solve the problem you have to explicitly URL encode the product number (for example with encodeURIComponent(...)).
This error will probably also occur on calling other endpoints if special characters are used.

**To Reproduce**

1. Using product number 'DEV?TEST06'
2. Calling GET product endpoint for product 'DEV?TEST06' (Product.requestBuilder().getByKey('DEV?TEST06'))
3. Error Message: 'Invalid URI segment'

**Expected behavior**
The product number must be URL encoded automatically.

**Used Versions:**

- node version via `node -v`: 12.22.7
- npm version via `npm -v`: 6.14.15
- SAP Cloud SDK version you used as dependency: 1.53.0

**Code Examples**
```
const query = Product.requestBuilder().getByKey('DEV?TEST06');
query.execute(...);
```

**Log file**
ErrorWithCause: OData get by key request failed!
    at new ErrorWithCause (...\node_modules\@sap-cloud-sdk\util\src\error-with-cause.ts:14:5)
    at ...\node_modules\@sap-cloud-sdk\core\src\odata-common\request-builder\get-by-key-request-builder.ts:76:15
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
Caused by:
ErrorWithCause: get request to .../sap/opu/odata/sap/API_PRODUCT_SRV failed! Invalid URI segment
{"code":"00505692409C1EE98BD4AA91DC38491E","message":{"lang":"en","value":"Invalid URI segment"},"innererror":{"transactionid":"DA22792F83AA0220E0061B8B6F009BC1","timestamp":"20211215131053.1522770","Error_Resolution":{"SAP_Transaction":"For backend administrators: use ADT feed reader \"SAP Gateway Error Log\" or run transaction /IWFND/ERROR_LOG on SAP Gateway hub system and search for entries with the timestamp above for more details","SAP_Note":"See SAP Note 1797736 for error analysis (https://service.sap.com/sap/support/notes/1797736)"}}}
    at new ErrorWithCause (...\node_modules\@sap-cloud-sdk\util\src\error-with-cause.ts:14:5)
    at constructError (...\node_modules\@sap-cloud-sdk\core\src\odata-common\request\odata-request.ts:270:12)
    at ...\node_modules\@sap-cloud-sdk\core\src\odata-common\request\odata-request.ts:240:13
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
Caused by:
Error: Request failed with status code 400
    at createError (...\node_modules\axios\lib\core\createError.js:16:15)
    at settle (...\node_modules\axios\lib\core\settle.js:17:12)
    at IncomingMessage.handleStreamEnd (...\node_modules\axios\lib\adapters\http.js:269:11)
    at IncomingMessage.emit (events.js:326:22)
    at endReadableNT (_stream_readable.js:1241:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)

**Impact / Priority**

This bug was noticed during our internal tests. However, this error can occur at any time for our customers.

**Comment by florian-richter** [comment]
Hi @jwe-cdn,

thanks for the report. I checked our code and you seem to be correct. As a short term workaround, you could run `encodeURIcomponent` yourself before passing the ID.

We are currently building version 2 of the SAP Cloud SDK for JavaScript and will try to publish a fix shortly after version 2 is released. Since we are not planning to backport any changes, it will likely not get fixed on version 1.

**Comment by marikaner** [comment]
Related to #1398.

**Comment by deekshas8** [comment]
Hi @jwe-cdn,

We have fixed the bug in our latest released v2.1.0. Please update to test it out. 
I'll close the ticket for now. If you experience problems, please reopen.
